x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/10_basic.yml

```diff
@@ -50,29 +50,33 @@ setup:
   - match: { data_streams.0.timestamp_field.name: '@timestamp' }
   - match: { data_streams.0.generation: 1 }
   - length: { data_streams.0.indices: 1 }
-  - match: { data_streams.0.indices.0.index_name: '.ds-simple-data-stream1-000001' }
+  - match: { data_streams.0.indices.0.index_name: '/\.ds-simple-data-stream1-(\d{4}-\d{2}-\d{2}-)?000001/' }
   - match: { data_streams.0.status: 'GREEN' }
   - match: { data_streams.0.template: 'my-template1' }
   - match: { data_streams.0.hidden: false }
   - match: { data_streams.1.name: simple-data-stream2 }
   - match: { data_streams.1.timestamp_field.name: '@timestamp' }
   - match: { data_streams.0.generation: 1 }
   - length: { data_streams.1.indices: 1 }
-  - match: { data_streams.1.indices.0.index_name: '.ds-simple-data-stream2-000001' }
+  - match: { data_streams.1.indices.0.index_name: '/\.ds-simple-data-stream2-(\d{4}-\d{2}-\d{2}-)?000001/' }
   - match: { data_streams.1.template: 'my-template2' }
   - match: { data_streams.0.hidden: false }
 
+  # save the backing index names for later use
+  - set: { data_streams.0.indices.0.index_name: idx0name }
+  - set: { data_streams.1.indices.0.index_name: idx1name }
+
   - do:
       indices.get_mapping:
-        index: .ds-simple-data-stream1-000001
+        index: .ds-simple-data-stream1-*000001
         expand_wildcards: hidden
-  - match: { \.ds-simple-data-stream1-000001.mappings.properties.@timestamp.type: 'date' }
+  - match: { $idx0name.mappings.properties.@timestamp.type: 'date' }
```

Comment:
I'm not sure why the test fails on this line when the near-identical line below succeeds:

https://github.com/elastic/elasticsearch/pull/65205/files#diff-bb60a4ca6d613aa0f52c15e7278930763876d350d28af32297f2dac488316798R112

Reply:
iirc `index: .ds-simple-data-stream1-*000001` doesn't work, because wildcard has to be at the end,
this `index: $idx0name` should work.

Reply:
Making this change and then it fails with: `no object found for [mappings] within object of class [class java.lang.String]` at line 73. I think it can't select the right json field in the response body, because the `$idx0name` variable contains dots in its name.

Reply:
I don't think variables like `$idx0name` can be used as field name, because it contains dot, which the yaml testing code interprets as a nested json field. I don't think there is escaping logic?

Reply:
> I don't think variables like `$idx0name` can be used as field name, because it contains dot, which the yaml testing code interprets as a nested json field. I don't think there is escaping logic?

Using variables as part of a field name in match clauses appears to work in other places such as the link above. The only difference I saw between the linked example and what I tried here is that here the variable is at the beginning of the field name rather than somewhere in the middle. I don't know if that matters, but I don't think that it should matter.